torch/_dynamo/output_graph.py

```diff
@@ -814,18 +865,53 @@ def remove_unused_graphargs(self) -> None:
                 elif node.op == "call_function" and node.target is operator.getitem:
                     self.remove_node(node)
 
-        for i, node in enumerate(self.placeholders):
-            if not node.users:
-                log.debug(
-                    "REMOVE UNUSED GRAPHARG %s", node.meta["grapharg"].source.name()
-                )
-                # I'm not really sure why you need to delete these from the
-                # node since the node is going to get removed
-                if "example_value" in node.meta:
-                    del node.meta["example_value"]
-                del node.meta["grapharg"]
-                self.remove_node(node)
-                self.real_value_cache.pop(node, None)
+        def placeholder_binds_symbol(node):
+            arg = node.meta["grapharg"]
```

Comment:
nit: you do `["grapharg"]` in enough places across a few files to warrant giving it a const imo

Reply:
We don't const "val" or "example_input" though?

Reply:
No no, I mean, the string! Like, just do something like

META_GRAPHARG_KEY = "grapharg"

nit anyway.

Reply:
I understood you. We don't have a key like this for "example_input". Are you saying we should have a key for that too?